lib/rubocop/cop/factory_bot/association_style.rb

```diff
@@ -218,6 +224,14 @@ def options_for_autocorrect_to_implicit_style(node)
           end
           options
         end
+
+        def trait_within_trait?(node)
+          factory_node = node.each_ancestor(:block).find do |ancestor|
+            ancestor.send_node.method?(:factory)
```

Comment:
I’m worried about the time complexity of this.

Reply:
The logic is modified for https://github.com/rubocop/rubocop-factory_bot/pull/59#discussion_r1292613483 . Please point out if there is another way to do it better.